app/deployments/routes.py

```diff
@@ -437,7 +446,9 @@ def get_sec_groups(conn, server_id, public=True):
 
     if public:
         return_sec_group_list = [
-            sec_group for sec_group in return_sec_group_list if search(substring, sec_group["name"])
```

Comment:
This and the previous part are a bit convoluted.
First, the terms `sec_group`, `sec_group_list` and `return_sec_group_list` are not clear names.
I suggest to use simply `sec_group` for the single entity and `sec_groups` to identify the list.
Then, the `return_sec_groups` list is most probably not necessary.

More comments follow below

Reply:
```python
return_sec_group_list = [
    sec_group for sec_group in return_sec_group_list if search(substring, sec_group["name"])
    for sec_group in return_sec_group_list
     if search(substring, sec_group["name"])
]
```

This is very convoluted too. First: is regex search `really needed`? We are looking for are simple substring, not a real regex pattern. Before all, rename `substring` to something more clear, like for example `pub_network_key`.
Then, since we are filtering all sec groups that has such key, consider using the `filter` function.

```python

if public:
   return list(filter(lambda sec_group: "pub_network" in sec_group["name"]))
```